packages/myst-cli/src/process/site.ts

```diff
@@ -253,6 +261,19 @@ export function selectPageReferenceStates(
     })
     .filter((state): state is ReferenceState => !!state);
   if (!opts?.suppressWarnings) warnOnDuplicateIdentifiers(session, pageReferenceStates);
+  pageReferenceStates.forEach((state) => {
+    const { mdast } = cache.$getMdast(state.filePath)?.post ?? {};
+    if (!mdast) return;
+    const vfile = new VFile();
+    vfile.path = state.filePath;
+    buildIndexTransform(
```

Comment:
This transform must occur after `processMdast` but before the parallel `postProcessMdast` step... For now, that is this overloaded `selectPageReferenceStates` function.

I think the complexity of all these different steps will be simplified once we tackle https://github.com/jupyter-book/mystmd/issues/1351